src/main/kotlin/io/github/yuutoproject/yuutobot/commands/minigame/MinigameInstance.kt

```diff
@@ -0,0 +1,182 @@
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.yuutoproject.yuutobot.commands.minigame
+
+import io.github.yuutoproject.yuutobot.commands.Minigame
+import io.github.yuutoproject.yuutobot.objects.Question
+import net.dv8tion.jda.api.EmbedBuilder
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionAddEvent
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionRemoveEvent
+import java.awt.Color
+
+class MinigameInstance(
+    private val questions: MutableList<Question>,
+    // The ID of the channel the game is in
+    // Effectively the ID of the game instance itself
+    val id: Long,
+    // This is necessary for when the game is finished and we want to remove the object
+    private val manager: Minigame,
+    private val maxRounds: Int
+) {
+    // Is the game in progress or not?
+    var begun = false
+
+    // String ID of user and their scores
+    var players = mutableMapOf<Long, Int>()
+    var timer = System.currentTimeMillis()
+
+    private var rounds = 1
+
+    private lateinit var startingMessageID: String
+
+    private lateinit var currentQuestion: Question
+    private lateinit var currentAnswers: MutableList<String>
+
+    fun start(event: GuildMessageReceivedEvent) {
+        event.channel.sendMessage("Starting a game with $maxRounds rounds...").queue()
+
+        val embed = EmbedBuilder()
+            .setColor(Color.BLUE)
+            .setTitle("Minigame Starting!")
+            .setDescription(
+                "React below to join the game! \n" +
+                    "This game may contain spoilers or NSFW themes.\n" +
+                    "Please run `minigame skip` in order to skip a question."
+            )
+        val startingMessage = event.channel.sendMessage(embed.build()).complete()
+        startingMessageID = startingMessage.id
+
+        startingMessage.addReaction("U+1F1F4").complete()
+
+        for (countdown in 10 downTo 0 step 2) {
+            embed.setDescription(
+                "React below to join the game!\n" +
+                    "This game may contain spoilers or NSFW themes.\n" +
+                    "Please run `minigame skip` in order to skip a question.\n" +
+                    "Current players: ${getPlayers()}\n" +
+                    "$countdown seconds left!"
+            )
+            startingMessage.editMessage(embed.build()).complete()
+
+            Thread.sleep(2000)
```

Comment:
Don't do this, this blocks any new commands from executing
Have a look at this https://kotlinlang.org/docs/reference/coroutines/basics.html